### ISSUE TITLE
Make a script/web extension that allows for simple mod.io url integration 

### ISSUE THREAD

**Issue by AichiChikuwa** [trumank/drg-mod-integration#107]
possible implementation:
- the script/extension adds a button to the mod.io mod page.
- when the script/extension's button is clicked, the url with a special identification string (may be something like "extension_url_link: ...") is copied to the clipboard.
- the app tracks the clipboard, and when it finds out that the clipboard contains that identification string, it gets the url and do the `add mod` thingy.
- *since the solution utilizes user's clipboard, the script/extension would probably need a one-time warning to let the user know that their clipboard is being tracked and will replace its content when the button is clicked.

**Comment by trumank** [comment]
Monitoring the clipboard is pretty invasive and provides little benefit as you can easily paste anywhere into the window with ctrl+v (a feature that admittedly needs to be made more discoverable.) Adding a custom URL protocol I think is the way to go.

I've been tinkering with expanding the integration to cover other Unreal Engine games such as Payday 3 which uses [Mod Workshop](https://modworkshop.net/) as its primary mod provider. They have a proposal to add a `mws://` protocol which will launch the external mod manager. A browser extension could be written for mod.io to add a similar URL to the mod page.

[Discussion in the MWS discord](https://discord.com/channels/242285927056015361/1162868928108830861):
> The user will see a new button next to the download button that, upon clicking, will send the user to a special link that is to be defined by the person adding it. The site will provide special parameters to let mod managers get what they need to install the mod.
> 
> for example:
> manager_name - The name of the mod manager.
> mod_id - The id of the mod
> file_id - The id of the file that's requested to install.
> game_id - The id of the game. For example, pd2 uses 1.
> 
> With possibly more if needed.
> So in the end if you define something like:
> 
> mws://:manager_name/install/:mod_id/:file_id
> You'd get:
> mws://moolahmodlauncher/install/12345/67890
> 
> This will let the moolah mod launcher the heads up that a file needs to be installed of mod 12345 with file ID 6789.
> ![image](https://github.com/trumank/drg-mod-integration/assets/1144160/bf34fa78-1e50-4ee3-b7c5-fd4ffc57bf37)

**Comment by AichiChikuwa** [comment]
That definitely sounds like a better idea!
Just to clarify: instead of making a DRG mod manager, this repo is likely to be A MOD MANAGER like the Vortex app to nexusmods.com right?

**Comment by trumank** [comment]
Yes, I want to pivot to more than just DRG, but still limited to Unreal Engine games (since they are modded in very similar ways). There's also backend support for multiple mod providers, unlike Vortex (primarily mod.io for now but I have a work in progress provider for MWS.)
